### PR TITLE
Improve code

### PR DESCRIPTION
**Pull request by javnov**
- Add CacheContentDownloader class to cache resources content and avoid unnecessary request
- Move Injector methods to an extension class
- Improve code readability

### CODE REVIEW

**Comment by javnov** [comment]
@jdmesalosada check this, and let me know your comments

File: javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        public string ToJson()
+        {
+            return JsonConvert.SerializeObject(this, jsonSerializerSettings);
+        }
+
+        private void InitializeIncludeList()
+        {
+            if (_includeList == null)
+            {
+                lock (syncObject)
```
Comment (jdmesalosada) [inline javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs]: Is neccessary to use this lock?

File: javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        /// <summary>
+        /// Get first selector of the first entry on include list
+        /// </summary>
+        /// <returns></returns>
+        public string GetFirstItemToInclude()
+        {
+            if (_includeList == null || _includeList.Count == 0)
+                throw new InvalidOperationException("You must add at least one selector to include");
+
+            return _includeList.First().First();
```
Comment (jdmesalosada) [inline javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs:75]: Here you are validating that the _includeList has not empty values but what about the second First?

Comment (javnov) [reply javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs:75]: The thing is the JS implementation is an array of arrays, but in the java version they are using a simple array of strings. Maybe we have to select only the first entry not the first selector of the fisrt entry.

File: javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs (hunk 1 above)
Comment (javnov) [reply javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs]: nope.

File: javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs (hunk 2 above)
Comment (javnov) [reply javnov.Selenium.Axe/javnov.Selenium.Axe/IncludeExcludeManager.cs:75]: @jdmesalosada what do you think ?

**Review by jdmesalosada — APPROVED** [review]

**Pull request merged**